mro/__init__.py

```diff
@@ -35,96 +35,120 @@ def execute_sql(sql, values=None):
         for foreign_key in cursor2:
             foreign_keys[foreign_key[0]] = (foreign_key[1], foreign_key[2])
 
-        # get foreign keys
-        cursor2.execute(f"""select  
-            kcu1.table_name as fk_table_name
-            ,kcu1.column_name as fk_column_name
-            ,kcu2.column_name as referenced_column_name 
-        from information_schema.referential_constraints as rc 
-
-        inner join information_schema.key_column_usage as kcu1 
-            on kcu1.constraint_catalog = rc.constraint_catalog  
-            and kcu1.constraint_schema = rc.constraint_schema 
-            and kcu1.constraint_name = rc.constraint_name
-
-        inner join information_schema.key_column_usage as kcu2 
-            on kcu2.constraint_catalog = rc.unique_constraint_catalog  
-            and kcu2.constraint_schema = rc.unique_constraint_schema 
-            and kcu2.constraint_name = rc.unique_constraint_name 
-            and kcu2.ordinal_position = kcu1.ordinal_position
-            and kcu2.table_name = '{table_name}';""")
+        # Get foreign keys (part 2)
+        # https://dba.stackexchange.com/a/218969
+        cursor2.execute(f"""
+            select 
+                 tbl.relname
+                ,col.attname
+                ,fcol.attname
+            from pg_catalog.pg_constraint con
+            join lateral unnest(con.conkey) with ordinality as u(attnum, attposition) on true
+            join pg_class tbl on tbl.oid = con.conrelid
+            join pg_attribute col on (col.attrelid = tbl.oid and col.attnum = u.attnum)
+            join lateral unnest(con.confkey) with ordinality as fu(attnum, attposition) on true
+            join pg_class ftbl on ftbl.oid = con.confrelid
+            join pg_attribute fcol on (fcol.attrelid = ftbl.oid and fcol.attnum = fu.attnum)
+            where 
+                    con.confrelid = '{table_name}'::regclass 
+                and con.contype = 'f';        
+        """)
         connection.commit()
 
         foreign_key_targets = []
         for foreign_key in cursor2:
             foreign_key_targets.append((foreign_key[0], foreign_key[1], foreign_key[2]))
 
         # Get primary keys
-        cursor2.execute(f"""select column_name from information_schema.table_constraints tc
-            join information_schema.constraint_column_usage AS ccu USING (constraint_schema, constraint_name)
-            where constraint_type='PRIMARY KEY' and tc.table_name='{table_name}'""")
+        # https://wiki.postgresql.org/wiki/Retrieve_primary_key_columns
+        cursor2.execute(f"""
+            select
+                a.attname
+            from pg_index i
+            join pg_attribute a on a.attrelid = i.indrelid and a.attnum = any(i.indkey)
+            where  
+                    i.indrelid = '{table_name}'::regclass
+                and i.indisprimary;        
+        """)
         connection.commit()
         primary_key_columns = [row[0] for row in cursor2]
 
         # Get columns
```

Comment:
There are some things in `information_schema.columns` that I couldn't figure out how to access directly in Postgres. I've left it using `information_schema` but now only query the columns that we need, which still seems to make it considerably faster to run.